pkg/ruler/notifier.go

```diff
@@ -26,11 +26,13 @@ import (
 )
 
 type NotifierConfig struct {
-	TLS       tls.ClientConfig `yaml:",inline"`
-	BasicAuth util.BasicAuth   `yaml:",inline"`
+	TLSEnabled bool             `yaml:"tls_enabled" category:"advanced"`
+	TLS        tls.ClientConfig `yaml:",inline"`
+	BasicAuth  util.BasicAuth   `yaml:",inline"`
 }
 
 func (cfg *NotifierConfig) RegisterFlags(f *flag.FlagSet) {
+	f.BoolVar(&cfg.TLSEnabled, "ruler.tls-enabled", true, "Enable TLS for gRPC client connecting to alertmanager.")
```

Comment:
The CLI flag should be `-ruler.alertmanager-client.tls-enabled` to keep consistency with the other flags in the data structure (which are inlined). @hi-rustin can you fix it, please? /cc @dimitarvdimitrov

Reply:
good point. I missed this. I opened https://github.com/grafana/mimir/pull/3597 to correct the CLi flag

Reply:
Thanks for the fix!